components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -918,7 +905,133 @@ void AdsServiceImpl::MigratePrefsVersion2To3() const {
 }
 
 int AdsServiceImpl::GetPrefsVersion() const {
-  return profile_->GetPrefs()->GetInteger(prefs::kVersion);
+  return GetIntegerPref(prefs::kVersion);
+}
+
+bool AdsServiceImpl::GetBooleanPref(
```

Comment:
for consistancy I would suggest that we use same naming that @bridiver used in rewards for it `GetBooleanState`. So replace `*Pref` with `*State`

Reply:
Right now this change is not related to that refactor, as this change is more concerning an issue where registering a preference does not persist the default value. I will add comments to reflect this